packages/api/models/executions.js

```diff
@@ -76,7 +83,15 @@ class Execution extends Manager {
     const concurrencyLimit = process.env.CONCURRENCY || 10;
     const limit = pLimit(concurrencyLimit);
 
-    const updatePromises = oldExecutionRows.Items.map((row) => limit(() => this.update({ arn: row.arn }, {}, ['originalPayload', 'finalPayload'])));
+    const updatePromises = oldExecutionRows.Items.map((row) => limit(() => {
+      if (row.status === 'completed' && !disableComplete && row.updatedAt <= completeMaxMs) {
```

Comment:
This should get refactored into multiple DB query/filters once we're not using dynamo.   Avoiding rolling it into the query here to avoid scanning the table twice when both cleanups are both enabled.